scripts/tools/video_capture.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+"""
+Live estimation of burned calories.
+Usage:
+  video_capture.py --duration=DURATION
+               [--pre_recording_duration=PRE_RECORDING_DURATION]
+               [--number_videos=NUMBER_VIDEOS]
+               [--camera_id=CAMERA_ID]
+               [--path_out=PATH_OUT]
+               [--file_name=FILE_NAME]
+  video_capture.py (-h | --help)
+
+Options:
+  --duration=DURATION                                   Recording time in seconds
+  --pre_recording_duration=PRE_RECORDING_DURATION       Duration for the pre-recording period
+  --number_videos=NUMBER_VIDEOS                         Number of videos to record [default: 1]
+  --camera_id=CAMERA_ID                                 ID of the camera to stream from
+  --path_out=PATH_OUT                                   Videos folder to stream to [default: output/]
+  --file_name=FILE_NAME                                 Video filename, followed by {video_number}.mp4 [default: output]
+"""
+
+import numpy as np
+import os
+import cv2
+import time
+from docopt import docopt
+
+
+def _get_fps(video):
+    """
+    Helper method to calculate the rate of frames per second to be set later for recording and saving the videos.
+
+    :param video:           VideoCapture
+        Object for the connected camera source
+    :return recorded_fps:   int
+        The calculated FPS for the given camera source
+    """
+    number_frames = 60
+
+    # Calculate time taken (in seconds) to read `number_frames`
+    start_time = time.time()
+    for n in range(number_frames):
+        _, _ = video.read()
+    end_time = time.time()
+
+    calculated_fps = int(number_frames/(end_time - start_time))
+
+    return 16. if calculated_fps <= 16 else calculated_fps
```

Comment:
Is a float acceptable here?

Reply:
I didn't mean to put that as a float, but it didn't cause any issue. To be on a safer side, I'll change that to an integer.